tests/sweep_runner/test_param_sweep_runner.py

```diff
@@ -115,20 +120,90 @@ def tests_prepare_raises_error_if_too_many_sino_slices(mocker: MockerFixture):
         chunk_shape=GLOBAL_SHAPE,
         block_start=0,
     )
-    loader = make_test_loader(mocker, block=block)
-    m1 = make_test_method(mocker)
-    sweep_wrapper = make_test_method(mocker, param_1=(0,))
-    pipeline = Pipeline(loader=loader, methods=[m1, sweep_wrapper])
-    runner = ParamSweepRunner(pipeline, MPI.COMM_WORLD)
+    preview = PreviewConfig(
+        angles=PreviewDimConfig(start=0, stop=180),
+        detector_y=PreviewDimConfig(start=0, stop=100),
+        detector_x=PreviewDimConfig(start=0, stop=160),
+    )
 
-    with pytest.raises(ValueError) as e:
-        runner.prepare()
+    class FakeModule:
+        def paganin(data: np.ndarray, alpha: float, pixel_size: float, energy: float, dist: float):  # type: ignore
+            return data * alpha
 
-    err_str = (
-        "Parameter sweep runs support input data containing <= 7 sinogram slices, "
-        "input data contains 10 slices"
+    mocker.patch(
+        "httomo.method_wrappers.generic.import_module", return_value=FakeModule
     )
-    assert err_str in str(e)
+
+    # Stuff take from `make_test_loader()`
+    loader: LoaderInterface = mocker.create_autospec(
+        LoaderInterface,
+        instance=True,
+        pattern=Pattern.all,
+        method_name="testloader",
+        reslice=False,
+        preview=preview,
+    )
+
+    def mock_make_data_source(padding) -> DataSetSource:
+        ret = mocker.create_autospec(
+            DataSetSource,
+            global_shape=block.global_shape,
+            dtype=block.data.dtype,
+            chunk_shape=block.chunk_shape,
+            chunk_index=block.chunk_index,
+            slicing_dim=1 if loader.pattern == Pattern.sinogram else 0,
+            aux_data=block.aux_data,
+            preview=preview,
+        )
+        type(ret).raw_shape = mock.PropertyMock(return_value=GLOBAL_SHAPE)
+        slicing_dim: Literal[0, 1, 2] = 0
+        mocker.patch.object(
+            ret,
+            "read_block",
+            side_effect=lambda start, length: DataSetBlock(
+                data=block.data[start : start + length, :, :],
+                aux_data=block.aux_data,
+                global_shape=block.global_shape,
+                chunk_shape=block.chunk_shape,
+                slicing_dim=slicing_dim,
+                block_start=start,
+                chunk_start=block.chunk_index[slicing_dim],
+            ),
+        )
+        return ret
+
+    mocker.patch.object(
+        loader,
+        "make_data_source",
+        side_effect=mock_make_data_source,
+    )
+
+    # Create sweep method wrapper, passing the sweep values for the param to sweep over
+    NO_OF_SWEEPS = 2
```

Comment:
Please correct me if I'm wrong, but it seems like `NO_OF_SWEEPS` isn't being used anywhere in this test? If that's the case, could it be removed please.